### PR TITLE
[libc++][test] disable all atomic<long double> tests

### LINKED ISSUE #73791: [libcxx] New std/atomics/atomics.types.generic/atomics.types.float tests hang on amd64
These four tests introduced in de7fbfeef5984ed3dc178957fbfaf4b1bb95fa94 hang on my Gentoo amd64 system:

```
  llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/fetch_add.pass.cpp
  llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/fetch_sub.pass.cpp
  llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/operator.minus_equals.pass.cpp
  llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/operator.plus_equals.pass.cpp
```

I've killed them after they've kept the CPU 100% busy for 40 minutes. Please fix them or revert.

CC @huixie90 

<details>
<summary>Full output (after killing)</summary>

```
FAIL: llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/operator.minus_equals.pass.cpp (9682 of 9685)
******************** TEST 'llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/operator.minus_equals.pass.cpp' FAILED ********************
Exit Code: 241

Command Output (stdout):
--
# COMPILED WITH
/usr/lib/ccache/bin/x86_64-pc-linux-gnu-clang++ /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/std/atomics/atomics.types.generic/atomics.types.float/operator.minus_equals.pass.cpp -pthread --target=x86_64-pc-linux-gnu -nostdinc++ -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/support -std=c++26 -Werror -Wall -Wctad-maybe-unsupported -Wextra -Wshadow -Wundef -Wunused-template -Wno-unused-command-line-argument -Wno-attributes -Wno-pessimizing-move -Wno-noexcept-type -Wno-atomic-alignment -Wno-reserved-module-identifier -Wdeprecated-copy -Wdeprecated-copy-dtor -Wno-user-defined-literals -Wno-tautological-compare -Wsign-compare -Wunused-variable -Wunused-parameter -Wunreachable-code -Wno-unused-local-typedef -Wno-local-type-template-args -Wno-c++11-extensions -Wno-unknown-pragmas -Wno-pass-failed -Wno-mismatched-new-delete -Wno-redundant-move -Wno-self-move -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBCPP_ENABLE_EXPERIMENTAL -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_NONE -Werror=thread-safety -Wuser-defined-warnings -latomic -lc++experimental -nostdlib++ -L /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -Wl,-rpath,/var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -lc++ -o /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/operator.minus_equals.pass.cpp.dir/t.tmp.exe
# executed command: /usr/lib/ccache/bin/x86_64-pc-linux-gnu-clang++ /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/std/atomics/atomics.types.generic/atomics.types.float/operator.minus_equals.pass.cpp -pthread --target=x86_64-pc-linux-gnu -nostdinc++ -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/support -std=c++26 -Werror -Wall -Wctad-maybe-unsupported -Wextra -Wshadow -Wundef -Wunused-template -Wno-unused-command-line-argument -Wno-attributes -Wno-pessimizing-move -Wno-noexcept-type -Wno-atomic-alignment -Wno-reserved-module-identifier -Wdeprecated-copy -Wdeprecated-copy-dtor -Wno-user-defined-literals -Wno-tautological-compare -Wsign-compare -Wunused-variable -Wunused-parameter -Wunreachable-code -Wno-unused-local-typedef -Wno-local-type-template-args -Wno-c++11-extensions -Wno-unknown-pragmas -Wno-pass-failed -Wno-mismatched-new-delete -Wno-redundant-move -Wno-self-move -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBCPP_ENABLE_EXPERIMENTAL -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_NONE -Werror=thread-safety -Wuser-defined-warnings -latomic -lc++experimental -nostdlib++ -L /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -Wl,-rpath,/var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -lc++ -o /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/operator.minus_equals.pass.cpp.dir/t.tmp.exe
# EXECUTED AS
/usr/bin/python3.12 /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/utils/run.py --execdir /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/operator.minus_equals.pass.cpp.dir --  /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/operator.minus_equals.pass.cpp.dir/t.tmp.exe
# executed command: /usr/bin/python3.12 /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/utils/run.py --execdir /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/operator.minus_equals.pass.cpp.dir -- /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/operator.minus_equals.pass.cpp.dir/t.tmp.exe
# note: command had no output on stdout or stderr
# error: command failed with exit status: 241

--

********************
FAIL: llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/fetch_sub.pass.cpp (9683 of 9685)
******************** TEST 'llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/fetch_sub.pass.cpp' FAILED ********************
Exit Code: 241

Command Output (stdout):
--
# COMPILED WITH
/usr/lib/ccache/bin/x86_64-pc-linux-gnu-clang++ /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/std/atomics/atomics.types.generic/atomics.types.float/fetch_sub.pass.cpp -pthread --target=x86_64-pc-linux-gnu -nostdinc++ -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/support -std=c++26 -Werror -Wall -Wctad-maybe-unsupported -Wextra -Wshadow -Wundef -Wunused-template -Wno-unused-command-line-argument -Wno-attributes -Wno-pessimizing-move -Wno-noexcept-type -Wno-atomic-alignment -Wno-reserved-module-identifier -Wdeprecated-copy -Wdeprecated-copy-dtor -Wno-user-defined-literals -Wno-tautological-compare -Wsign-compare -Wunused-variable -Wunused-parameter -Wunreachable-code -Wno-unused-local-typedef -Wno-local-type-template-args -Wno-c++11-extensions -Wno-unknown-pragmas -Wno-pass-failed -Wno-mismatched-new-delete -Wno-redundant-move -Wno-self-move -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBCPP_ENABLE_EXPERIMENTAL -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_NONE -Werror=thread-safety -Wuser-defined-warnings -latomic -lc++experimental -nostdlib++ -L /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -Wl,-rpath,/var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -lc++ -o /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/fetch_sub.pass.cpp.dir/t.tmp.exe
# executed command: /usr/lib/ccache/bin/x86_64-pc-linux-gnu-clang++ /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/std/atomics/atomics.types.generic/atomics.types.float/fetch_sub.pass.cpp -pthread --target=x86_64-pc-linux-gnu -nostdinc++ -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/support -std=c++26 -Werror -Wall -Wctad-maybe-unsupported -Wextra -Wshadow -Wundef -Wunused-template -Wno-unused-command-line-argument -Wno-attributes -Wno-pessimizing-move -Wno-noexcept-type -Wno-atomic-alignment -Wno-reserved-module-identifier -Wdeprecated-copy -Wdeprecated-copy-dtor -Wno-user-defined-literals -Wno-tautological-compare -Wsign-compare -Wunused-variable -Wunused-parameter -Wunreachable-code -Wno-unused-local-typedef -Wno-local-type-template-args -Wno-c++11-extensions -Wno-unknown-pragmas -Wno-pass-failed -Wno-mismatched-new-delete -Wno-redundant-move -Wno-self-move -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBCPP_ENABLE_EXPERIMENTAL -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_NONE -Werror=thread-safety -Wuser-defined-warnings -latomic -lc++experimental -nostdlib++ -L /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -Wl,-rpath,/var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -lc++ -o /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/fetch_sub.pass.cpp.dir/t.tmp.exe
# EXECUTED AS
/usr/bin/python3.12 /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/utils/run.py --execdir /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/fetch_sub.pass.cpp.dir --  /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/fetch_sub.pass.cpp.dir/t.tmp.exe
# executed command: /usr/bin/python3.12 /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/utils/run.py --execdir /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/fetch_sub.pass.cpp.dir -- /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/fetch_sub.pass.cpp.dir/t.tmp.exe
# note: command had no output on stdout or stderr
# error: command failed with exit status: 241

--

********************
FAIL: llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/fetch_add.pass.cpp (9684 of 9685)
******************** TEST 'llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/fetch_add.pass.cpp' FAILED ********************
Exit Code: 241

Command Output (stdout):
--
# COMPILED WITH
/usr/lib/ccache/bin/x86_64-pc-linux-gnu-clang++ /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/std/atomics/atomics.types.generic/atomics.types.float/fetch_add.pass.cpp -pthread --target=x86_64-pc-linux-gnu -nostdinc++ -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/support -std=c++26 -Werror -Wall -Wctad-maybe-unsupported -Wextra -Wshadow -Wundef -Wunused-template -Wno-unused-command-line-argument -Wno-attributes -Wno-pessimizing-move -Wno-noexcept-type -Wno-atomic-alignment -Wno-reserved-module-identifier -Wdeprecated-copy -Wdeprecated-copy-dtor -Wno-user-defined-literals -Wno-tautological-compare -Wsign-compare -Wunused-variable -Wunused-parameter -Wunreachable-code -Wno-unused-local-typedef -Wno-local-type-template-args -Wno-c++11-extensions -Wno-unknown-pragmas -Wno-pass-failed -Wno-mismatched-new-delete -Wno-redundant-move -Wno-self-move -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBCPP_ENABLE_EXPERIMENTAL -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_NONE -Werror=thread-safety -Wuser-defined-warnings -latomic -lc++experimental -nostdlib++ -L /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -Wl,-rpath,/var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -lc++ -o /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/fetch_add.pass.cpp.dir/t.tmp.exe
# executed command: /usr/lib/ccache/bin/x86_64-pc-linux-gnu-clang++ /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/std/atomics/atomics.types.generic/atomics.types.float/fetch_add.pass.cpp -pthread --target=x86_64-pc-linux-gnu -nostdinc++ -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/support -std=c++26 -Werror -Wall -Wctad-maybe-unsupported -Wextra -Wshadow -Wundef -Wunused-template -Wno-unused-command-line-argument -Wno-attributes -Wno-pessimizing-move -Wno-noexcept-type -Wno-atomic-alignment -Wno-reserved-module-identifier -Wdeprecated-copy -Wdeprecated-copy-dtor -Wno-user-defined-literals -Wno-tautological-compare -Wsign-compare -Wunused-variable -Wunused-parameter -Wunreachable-code -Wno-unused-local-typedef -Wno-local-type-template-args -Wno-c++11-extensions -Wno-unknown-pragmas -Wno-pass-failed -Wno-mismatched-new-delete -Wno-redundant-move -Wno-self-move -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBCPP_ENABLE_EXPERIMENTAL -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_NONE -Werror=thread-safety -Wuser-defined-warnings -latomic -lc++experimental -nostdlib++ -L /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -Wl,-rpath,/var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -lc++ -o /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/fetch_add.pass.cpp.dir/t.tmp.exe
# EXECUTED AS
/usr/bin/python3.12 /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/utils/run.py --execdir /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/fetch_add.pass.cpp.dir --  /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/fetch_add.pass.cpp.dir/t.tmp.exe
# executed command: /usr/bin/python3.12 /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/utils/run.py --execdir /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/fetch_add.pass.cpp.dir -- /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/fetch_add.pass.cpp.dir/t.tmp.exe
# note: command had no output on stdout or stderr
# error: command failed with exit status: 241

--

********************
FAIL: llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/operator.plus_equals.pass.cpp (9685 of 9685)
******************** TEST 'llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/operator.plus_equals.pass.cpp' FAILED ********************
Exit Code: 241

Command Output (stdout):
--
# COMPILED WITH
/usr/lib/ccache/bin/x86_64-pc-linux-gnu-clang++ /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/std/atomics/atomics.types.generic/atomics.types.float/operator.plus_equals.pass.cpp -pthread --target=x86_64-pc-linux-gnu -nostdinc++ -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/support -std=c++26 -Werror -Wall -Wctad-maybe-unsupported -Wextra -Wshadow -Wundef -Wunused-template -Wno-unused-command-line-argument -Wno-attributes -Wno-pessimizing-move -Wno-noexcept-type -Wno-atomic-alignment -Wno-reserved-module-identifier -Wdeprecated-copy -Wdeprecated-copy-dtor -Wno-user-defined-literals -Wno-tautological-compare -Wsign-compare -Wunused-variable -Wunused-parameter -Wunreachable-code -Wno-unused-local-typedef -Wno-local-type-template-args -Wno-c++11-extensions -Wno-unknown-pragmas -Wno-pass-failed -Wno-mismatched-new-delete -Wno-redundant-move -Wno-self-move -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBCPP_ENABLE_EXPERIMENTAL -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_NONE -Werror=thread-safety -Wuser-defined-warnings -latomic -lc++experimental -nostdlib++ -L /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -Wl,-rpath,/var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -lc++ -o /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/operator.plus_equals.pass.cpp.dir/t.tmp.exe
# executed command: /usr/lib/ccache/bin/x86_64-pc-linux-gnu-clang++ /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/std/atomics/atomics.types.generic/atomics.types.float/operator.plus_equals.pass.cpp -pthread --target=x86_64-pc-linux-gnu -nostdinc++ -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/include/c++/v1 -I /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/test/support -std=c++26 -Werror -Wall -Wctad-maybe-unsupported -Wextra -Wshadow -Wundef -Wunused-template -Wno-unused-command-line-argument -Wno-attributes -Wno-pessimizing-move -Wno-noexcept-type -Wno-atomic-alignment -Wno-reserved-module-identifier -Wdeprecated-copy -Wdeprecated-copy-dtor -Wno-user-defined-literals -Wno-tautological-compare -Wsign-compare -Wunused-variable -Wunused-parameter -Wunreachable-code -Wno-unused-local-typedef -Wno-local-type-template-args -Wno-c++11-extensions -Wno-unknown-pragmas -Wno-pass-failed -Wno-mismatched-new-delete -Wno-redundant-move -Wno-self-move -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBCPP_ENABLE_EXPERIMENTAL -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_NONE -Werror=thread-safety -Wuser-defined-warnings -latomic -lc++experimental -nostdlib++ -L /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -Wl,-rpath,/var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/lib -lc++ -o /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/operator.plus_equals.pass.cpp.dir/t.tmp.exe
# EXECUTED AS
/usr/bin/python3.12 /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/utils/run.py --execdir /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/operator.plus_equals.pass.cpp.dir --  /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/operator.plus_equals.pass.cpp.dir/t.tmp.exe
# executed command: /usr/bin/python3.12 /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/libcxx/utils/run.py --execdir /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/operator.plus_equals.pass.cpp.dir -- /var/tmp/portage/sys-libs/libcxx-18.0.0_pre20231123/work/runtimes_build-abi_x86_64.amd64/test/std/atomics/atomics.types.generic/atomics.types.float/Output/operator.plus_equals.pass.cpp.dir/t.tmp.exe
# note: command had no output on stdout or stderr
# error: command failed with exit status: 241

--

********************
********************
Failed Tests (4):
  llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/fetch_add.pass.cpp
  llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/fetch_sub.pass.cpp
  llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/operator.minus_equals.pass.cpp
  llvm-libc++-shared.cfg.in :: std/atomics/atomics.types.generic/atomics.types.float/operator.plus_equals.pass.cpp


Testing Time: 3378.31s

Total Discovered Tests: 9685
  Unsupported      :  571 (5.90%)
  Passed           : 9069 (93.64%)
  Expectedly Failed:   41 (0.42%)
  Failed           :    4 (0.04%)
```
</details>

### PR DESCRIPTION
**Pull request by huixie90**
fixes #73791

### CODE REVIEW

**Comment by EricWF** [comment]
This does not fix the issue, it hides it.

**Review by EricWF — CHANGES_REQUESTED** [review]
This is not an OK way to deal with broken tests.

**Comment by huixie90** [comment]
> This is not an OK way to deal with broken tests.
There are lots of discussions on the other PR when these tests were added.  Basically long double atomic is broken for years and we had some bug reports (against clang) open already. These tests were newly added which already had lots of platforms disabled. But after the merge, we just got more reports that these tests failed in the wild that lots of other platforms that are not in our CI. After some discussions, we will just disable the tests. atomic long double was and is broken. We need to fix the issue before we can have these tests running. 

We cannot fix it in the library side by our own. We need clang’s support (at least provide a way to zero out the long double’s 56bits padding)

**Review by ldionne — APPROVED** [review]
I agree with @huixie90 here. This is not fixing the broken atomic support, but we have other GH issues tracking that. This is merely making the CI green again for some configurations that are currently failing.

The bottom line is that atomic's support for `long double` is broken all around and we added tests for it, which now obviously fail. We should probably have punted on adding the tests for `long double` until they actually worked on all platforms, and this is what this PR is doing.

**Comment by vitalybuka** [comment]
Would it be better to disable `long double` test for x86_64 only?

File: libcxx/test/libcxx/atomics/atomics.types.generic/atomics.types.float/lockfree.pass.cpp [hunk 1]
```diff
@@ -45,7 +45,8 @@ void test() {
 int main(int, char**) {
   test<float>();
   test<double>();
-  test<long double>();
+  // https://github.com/llvm/llvm-project/issues/47978
```
Comment (mordante) [inline libcxx/test/libcxx/atomics/atomics.types.generic/atomics.types.float/lockfree.pass.cpp:48]:
```suggestion
  // TODO https://github.com/llvm/llvm-project/issues/47978
```
This makes it easier to find them.

**Review by mordante — COMMENTED** [review]
I also prefer to temporary disable the tests to get the CI green again.

**Review by ldionne — APPROVED** [review]
LGTM. I am going to merge this since CI seems happy and this will resolve CI issues that have been ongoing for way too long on Gentoo.

**Pull request merged**